gnucash/gnome/gnc-plugin-page-register.c

```diff
@@ -2659,6 +2669,114 @@ gnc_plugin_page_register_filter_response_cb (GtkDialog *dialog,
     LEAVE(" ");
 }
 
+static gchar*
+gpp_get_cleared_match_filter_text (cleared_match_t match, cleared_match_t match_it, const gchar* text)
+{
+    gchar *text_do_not = _("Do not show");
+    gchar *text_show = _("Show");
+
+    if ((match & match_it) == match_it)
+        return g_strdup_printf ("%s %s", text_show, text);
+    else
+        return g_strdup_printf ("%s %s", text_do_not, text);
+}
+
+static void
+gnc_plugin_page_register_set_filter_tooltip (GncPluginPageRegister *page)
+{
+    GncPluginPageRegisterPrivate *priv;
+    GncPluginPage *plugin_page;
+    gchar *text = NULL;
+    gchar *text_header = g_strdup_printf ("%s\n", _("Filter By:"));
+    gchar *text_start = NULL;
+    gchar *text_end = NULL;
+    gchar *text_cleared = NULL;
+
+    g_return_if_fail(GNC_IS_PLUGIN_PAGE_REGISTER(page));
+
+    ENTER(" ");
+    priv = GNC_PLUGIN_PAGE_REGISTER_GET_PRIVATE(page);
+    plugin_page = GNC_PLUGIN_PAGE(page);
+
+    // filtered start time
+    if (priv->fd.start_time != 0)
+    {
+        gchar *sdate = qof_print_date (priv->fd.start_time);
+        text_start = g_strdup_printf ("%s %s", _("Start Date:"), sdate);
+        g_free (sdate);
+    }
+    // filtered end time
+    if (priv->fd.end_time != 0)
+    {
+        gchar *edate = qof_print_date (priv->fd.end_time);
+        text_end = g_strdup_printf ("%s %s", _("End Date:"), edate);
+        g_free (edate);
+    }
+    // filtered match items
+    if (priv->fd.cleared_match != 31)
+    {
+        gchar *text_u = gpp_get_cleared_match_filter_text (priv->fd.cleared_match, 0x01, _("Unreconciled"));
+        gchar *text_c = gpp_get_cleared_match_filter_text (priv->fd.cleared_match, 0x02, _("Cleared"));
+        gchar *text_r = gpp_get_cleared_match_filter_text (priv->fd.cleared_match, 0x04, _("Reconciled"));
+        gchar *text_f = gpp_get_cleared_match_filter_text (priv->fd.cleared_match, 0x08, _("Frozen"));
+        gchar *text_v = gpp_get_cleared_match_filter_text (priv->fd.cleared_match, 0x10, _("Voided"));
+        text_cleared = g_strconcat (text_u, "\n", text_c, "\n", text_r, "\n", text_f, "\n", text_v, NULL);
```

Comment:
Let's try to shorten this output. For example make two lines, one with "Show a, b, d" and the second with "Hide c, e, f"

Reply:
Will have a look at trying this format, maybe put the dates on one line also ?

Reply:
Sure, you can try that.